core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -61,7 +61,7 @@ public class OAuth2Manager extends RefreshingAuthManager {
   private RESTClient refreshClient;
   private long startTimeMillis;
   private OAuthTokenResponse authResponse;
-  private AuthSessionCache sessionCache;
+  private AuthManagerSessionCache<String, OAuth2Util.AuthSession> sessionCache;
```

Comment:
Don't you want this to be `AuthManagerSessionCache<Object, OAuth2Util.AuthSession>`?  I don't think we need to preserve the string generic or maybe I'm misunderstanding what the end state is.  If we don't have the new protected `newAuthSessionCache` return a wider generic, we'll have to go through another deprecation cycle.

Reply:
The idea was to allow _other_ manager impls to use a different cache key. But this auth manager should keep using string keys imho. It's not the intent of this PR to change that.